File: Daily Test/Python/Split Palindromes 16.1.2026.py
def is_pal(s):
    i, j = 0, len(s) - 1
    while i < j:
        if s[i] != s[j]:
            return False
        i += 1
        j -= 1
    return True

s = input().strip()

for i in range(1, len(s)):
    left = s[:i]
    right = s[i:]
    if is_pal(left) and is_pal(right):
        print(left)
        print(right)
        break

File: Daily Test/Java/Split Palindromes 16.1.2026.java
import java.util.*;

public class Hello {

    static boolean isPal(String s) {
        int i = 0, j = s.length() - 1;
        while (i < j) {
            if (s.charAt(i) != s.charAt(j))
                return false;
            i++;
            j--;
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();

        for (int i = 1; i < str.length(); i++) {
            String left = str.substring(0, i);
            String right = str.substring(i);

            if (isPal(left) && isPal(right)) {
                System.out.println(left);
                System.out.println(right);
                break;
            }
        }
    }
}